core/src/main/java/org/elasticsearch/index/query/support/InnerHitBuilder.java

```diff
@@ -202,7 +200,7 @@ public String getName() {
     }
 
     public InnerHitBuilder setName(String name) {
-        this.name = name;
+        this.name = Objects.requireNonNull(name);
```

Comment:
Name is optional, if the name isn't specified it will default to either nestedPath or parentChildType.
I'll add a constructor that accepts either a nestedPath or a parentChildType.